Update manager.html: Adds "Join Server" button

y1337x opened the pull request:
Adds a button to join the server defined by a domain and port. Sometimes you cannot find the server because steam uses IP addresses. With this button steam will start if it is not running, starts ark and tries to connect to the server. To do that the domain is translated to an IP address.

Commit:
Update manager.html

Adds a button to join the server defined by a domain and port. Sometimes you cannot find the server because steam uses IP addresses. With this button steam will start if it is not running, starts ark and tries to connect to the server. To do that the domain is translated to an IP address.

skanatiker commented:
@DanyelSuxdorf : I think the configurability would be great here too. The public IP or DNS host name and the Port should be configurable in the preferences screen. Users should not have to modify html or javascript files manually.

y1337x commented:
I will make it configurable when I have time for that, maybe next weekend.

y1337x commented:
@skanatiker, can you help me? Web programming is pretty new for me. The button isn't working, I think the function isn't called and I don't know why.
 **manager.html**:
`<button type="button" class="btn btn-primary" style="text-align:center; width:100%" title="Opens Steam and tries to connect to the configurated host" ng-click="connectToServer()">Play on Server</button>` 

new function in **models/command.js**
`exports.connectToServer = function(wss) {
	configs.findByKey('server', function(err, serverConfig) {
		var address = "steam://connect/" + serverConfig.data.serverHost + ":27015";
		wss.broadcast('log', {log: "call steam://connect/" + serverConfig.data.serverHost + ":27015"});
		window.location.href = address;
	});
}`

added scope in **src/scripts/server/server.ctrl.js**
`$scope.connectToServer = function(){
        ws.send(JSON.stringify({event: "connectToServer"}));
    };`

added in **public/scripts/ark.js**
`e.connectToServer=function(){l.send(JSON.stringify({event:"connectToServer"}))}`

The host is added to the db and is loading correctly in the web interface.
